### DISCUSSION TITLE
no available capacity

### DISCUSSION THREAD

**Discussion by flosigime** [sqlpage/SQLPage#441] [Q&A] [answered]
Hello everyone, 

I have this warning in the sqlpage promp : [2024-06-21T20:06:11.310Z ERROR sqlpage::webserver::http] Could not flush data to client: no available capacity. 

I'm using a map component that gets a geojson of the geometry in a spatialite database but the map does not work. I tried to find geometric errors but everything seems in good health...

Thanks for your help

**Comment by lovasoa** [comment, marked as answer]
This error happens when your database emits a lot of data very quickly, and the web browser isn't able to download it fast enough. In this case, SQLPage will store the database's responses in memory while waiting for the client to download it. But it won't accept to store an arbitrarily large amount of data in memory, in order not to penalize other clients that may want to use the server at the same time. So, when the database has emitted more than 128  messages and the client hasn't downloaded them, sqlpage will throw an error. This limit of 128 is called `MAX_PENDING_MESSAGES`, and is currently hardcoded. I'll make it configurable in the next version so that you can increase it if you want.

But, knowing your application, I think the underlying problem is that you are trying to load a lot of data into a single map. What we could ideally do in SQLPage is to allow the map component to load data "on-demand", so that only the data located in the part of the map that you are currently viewing is loaded.

**Reply by lovasoa** [reply]
I made the update. You'll now have access to `max_pending_rows` in the documentation: https://github.com/lovasoa/SQLpage/blob/main/configuration.md

**Comment by flosigime** [comment]
Thanks, 

what's strange, it's that the map works well on a page, but not on another (ok for emprise, not for fait). And the amount of data in my case is not so big (221 polygon geometries). Really less than Noisy. 
Il look at the documentation, thanks.

**Reply by flosigime** [reply]
i suppose i can use `max_pending_rows` in the json, like `"max_pending-rows":"128"`

**Reply by lovasoa** [reply]
It will be `"max_pending_rows": 1024` for instance. But you'll have to wait until v0.24, which I'll release tomorrow if everything goes according to plan.

**Reply by flosigime** [reply]
i'm not hurry, don't worry... ;-)

**Comment by flosigime** [comment]
ok, despite the 0.24 version, the problem was the same. The map component refuses to load, but there was no message in the prompt. 
Thomas discovered it was because of the list component before the map component in the page, and especially the search parameter when there's nothing to view...

**Reply by lovasoa** [reply]
Hello ! Could you please make a detailed bug report for the problem you are seeing with 0.24 ? You can open a separate issue if this one about `no available capacity` is solved. Ideally if you could provide a precise dataset and `.sql` file to reproduce the issue, it would help fixing it for everyone.
